runtime/v2/manager.go

```diff
@@ -494,6 +494,12 @@ func (m *TaskManager) Delete(ctx context.Context, taskID string) (*runtime.Exit,
 	})
 
 	if err != nil {
+		if errdefs.IsDeadlineExceeded(err) {
+			dctx, cancel := timeout.WithContext(context.Background(), cleanupTimeout)
+			defer cancel()
+			shimTask.Shutdown(dctx)
+			shimTask.Client().Close()
```

Comment:
The timeout does not mean that shim's Delete method will execute the other logic correctly.

I don't think we could perform the cleanup operation after timeout.

Reply:
I would tend to agree with @Iceber on this.
Retrying this should be handled by the client (or cri).